pkg/util/kubernetes/kubelet/testdata/podlist_null_pod.json

```diff
@@ -0,0 +1,10 @@
+{
+    "kind": "PodList",
+    "apiVersion": "v1",
+    "metadata": {},
+    "items": [
```

Comment:
Before doing the filtering, is it worth seeing if there is a setting we can configure on the JSON deserializer to omit nil values?

Reply:
@kevinconaway I tried to look into it but didn't find an easy way with https://github.com/json-iterator/go 😞

Reply:
Got it.  The only other thing that I could think of would be to make `PodList.Items` be a `[]Pod` instead of a `[]*Pod`.  I'm not sure if there is a reason it needs to be a pointer but if not, that should fix it.

Reply:
Indeed not sure why it was a pointer in the first place as it's supposed to be read-only, maybe for memory efficiency, tried to look into this change as well but it's quite a big change actually, the effort to try to see the memory impact was not worth it IMHO